view/frontend/web/internals/common.js

```diff
@@ -443,6 +443,8 @@ requirejs(['algoliaBundle'], function(algoliaBundle) {
 		$(algoliaConfig.autocomplete.selector).each(function () {
 			$(this).closest('form').submit(function (e) {
 				var query = $(this).find(algoliaConfig.autocomplete.selector).val();
+				
+				query = query.replace(/#/g, '%23');
```

Comment:
As already proposed in #720, to be one hundred percent sure, we could also `encodeURI()` the complete search query, then all special characters would be encoded.

```
query = encodeURI(query);
query = query.replace(/#/g, '%23');
```

That would have the consequence if you search in german for example for "küchentuch #2349" you would have `?q=k%C3%BCchentuch%20%232349` instead of `?q=küchentuch%20%232349`.